apps/web/stores/AnswerStore.ts

```diff
@@ -87,19 +113,26 @@ export class AnswerStore {
   setAnswerValueOnChange = (value: AnswerTypes, questionId: string) => {
     this.eraseAnswersAfterCurrentItem();
 
-    this.rootStore.navigationStore.farthestItemId = questionId;
+    this.rootStore.navigationStore.setFarthestIds(questionId);
 
-    this.answers[questionId] = value;
+    this.setAnswerValue(
+      this.rootStore.navigationStore.currentWalkthroughId,
```

Comment:
Is it possible for this "id" to ever be out of sync?

Reply:
I don't think so. This only happens within the context of 1 question, so the current walkthrough id wouldn't change while a user is interacting with a question. And all state is updated before any rendering happens between questions.